src/Chord.hs

```diff
@@ -58,17 +76,19 @@ instance Show Accidental where
   show (AccFlat i) = concat $ replicate i "b"
   show AccNatural = ""
 
+nextNthNote :: Note -> Int -> Note
+nextNthNote note i = iterate succ note !! i
 
-nextNote :: Note -> Note
-nextNote G = A
-nextNote n = succ n
+prevNthNote :: Note -> Int -> Note
+prevNthNote note i = iterate pred note !! i
 
-prevNote :: Note -> Note
-prevNote A = G
-prevNote n = pred n
+data CChord = CChord Root Quality HighestNatural [Extension] Sus
+ deriving Show
 
-nextNthNote :: Note -> Int -> Note
-nextNthNote note i = iterate nextNote note !! i
+canonicalizeQuality :: Maybe Quality -> HighestNatural -> Quality
+canonicalizeQuality Nothing  (HighestNatural _ i) = if i < 7 then QMajor else QDominant
+canonicalizeQuality (Just q) _                    = q
 
-prevNthNote :: Note -> Int -> Note
-prevNthNote note i = iterate prevNote note !! i
+canonicalizeChord :: Chord -> CChord
+canonicalizeChord (Chord root mqual highNat exts sus) =
+  CChord root (canonicalizeQuality mqual highNat) highNat exts sus
```

Comment:
Canonical chords and 'normal' chords don't have a good reason to be separated at the module level; in fact, I think I'm going to change this even further to use `Chord` as the name of canonical chords, and `RawChord` as what the parser builds. This will clean up the code that uses `Chord` values.